src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/RNGCryptoServiceProvider.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+
+namespace System.Security.Cryptography
+{
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public sealed class RNGCryptoServiceProvider : RandomNumberGenerator
+    {
+        private readonly RandomNumberGenerator _impl;
+
+        public RNGCryptoServiceProvider() : this((CspParameters) null) { }
+        public RNGCryptoServiceProvider(string str) : this((CspParameters)null) { }
+        public RNGCryptoServiceProvider(byte[] rgb) : this((CspParameters)null) { }
+
+        public RNGCryptoServiceProvider(CspParameters cspParams)
+        {
+            if (cspParams != null)
+                throw new PlatformNotSupportedException();
+
+            // This class wraps RandomNumberGenerator.Create() from Algorithms assembly
+            _impl = Create();
+        }
+
+        public override void GetBytes(byte[] data) => _impl.GetBytes(data);
+
+        // GetBytes(data, offset, count) not forwarded here because the base just calls GetBytes(data)
```

Comment:
The base does, but the _impl doesn't... if we forward this to the _impl we get the lower-alloc gains.